mrmustard/physics/ansatz/polyexp_ansatz.py

```diff
@@ -208,6 +209,56 @@ def num_derived_vars(self) -> int:
     def num_vars(self):
         return self.A.shape[-1]
 
+    @property
+    def PS(self) -> PolyExpAnsatz:
+        r"""
+        The ansatz acting on real (i.e., phase-space) variables.
+        """
+        n = self.A.shape[-1]
+        if n % 2:
+            raise ValueError(
+                f"A phase space ansatz must have even number of indices. (n={n} is odd)"
+            )
+
+        if self.c.shape == ():
+            In = math.eye(n // 2, dtype=math.complex128)
+            W = math.block([[In, -1j * In], [In, 1j * In]]) / complex(
+                math.sqrt(2, dtype=math.complex128) * settings.HBAR
+            )
+
+            A = math.einsum("ji, ...jk, kl-> ...il", W, self.A, W)
+            b = math.einsum("ij, ...j-> ...i", W, self.b)
+            c = self.c / (2 * settings.HBAR) ** (n // 2)
+            return PolyExpAnsatz(A, b, c, lin_sup=self._lin_sup)
+
+        else:
+            if self.A.shape[-1] != 4:
+                raise ValueError(
+                    f"This transformation is written only for 2 core and 2 derived variables"
+                )
+            A_tmp = self.A
+
+            A_tmp = A_tmp[..., [0, 2, 1, 3], :]
+            A_tmp = A_tmp[..., [0, 2, 1, 3]]
+            b = self.b
+            b_tmp = b[..., [0, 2, 1, 3]]
+            c = c_in_PS(self.c)  # implements PS transformations on ``c``
```

Comment:
```suggestion
            A_tmp = A_tmp[..., [0, 2, 1, 3], :][..., [0, 2, 1, 3]]
            b = self.b[..., [0, 2, 1, 3]]
            c = c_in_PS(self.c)  # implements PS transformations on ``c``
```
saving a couple lines